azurerm/internal/services/network/parse/firewall_policy_test.go

```diff
@@ -0,0 +1,72 @@
+package parse
+
+import (
+	"testing"
+)
+
+func TestFirewallPolicyID(t *testing.T) {
+	testData := []struct {
+		Name   string
+		Input  string
+		Error  bool
+		Expect *FirewallPolicyId
+	}{
+		{
+			Name:  "Empty",
+			Input: "",
+			Error: true,
+		},
+		{
+			Name:  "No Resource Groups Segment",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000",
+			Error: true,
+		},
+		{
+			Name:  "No Resource Groups Value",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/",
+			Error: true,
+		},
+		{
+			Name:  "No Resource Groups Value",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/",
+			Error: true,
+		},
+		{
+			Name:  "No Policy Name",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Network/firewallPolicies",
+			Error: true,
+		},
+		{
+			Name:  "Correct Case",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Network/firewallPolicies/policy1",
+			Expect: &FirewallPolicyId{
+				ResourceGroup: "group1",
+				Name:          "policy1",
+			},
+		},
```

Comment:
Can we add a couple of more test cases here? Such as, Incorrect Case, Malformed Resource ID(out of order segments, extra segments, missing starting `/`, etc.)?

Reply:
Will do!